adb_graphics/datahandler/gribdata.py

```diff
@@ -481,128 +285,185 @@ class fieldData(UPPData):
+        flru = where((ceil > 0.0) & (ceil < 0.5), 3.01, flru)
+        flru = where((vis < 1.0), 3.01, flru)
 
         vis.close()
 
-        return xr.DataArray(flru)
+        return DataArray(flru)
 
     @property
     def cmap(self):
+        """
+        The LinearSegmentedColormap specified by the config key 'cmap'.
+        """
 
-        ''' Returns the LinearSegmentedColormap specified by the config key
-        "cmap" '''
-
-        return cm.get_cmap(self.vspec['cmap'])
+        return get_cmap(self.vspec["cmap"])
 
     @property
     def colors(self) -> np.ndarray:
-
-        '''
-        Returns a list of colors, specified by the config key "colors".
-
-        The yaml file "colors" key may contain a list or a function to be
-        called.
-        '''
-
-        color_spec = self.vspec.get('colors')
-
-        if isinstance(color_spec, (list, np.ndarray)):
-            return np.asarray(color_spec)
+        """
+        Returns an array of colors, specified by the config key "colors".
+        """
+
+        color_spec = self.vspec.get("colors", "")
+        if not color_spec:
+            msg = f"No colors definition found for {self.short_name} at {self.level}"
+            raise errors.NoGraphicsDefinitionForVariableError(msg)
         try:
             ret = self.__getattribute__(color_spec)
-            if callable(ret):
-                return ret()
-            return ret
-        except AttributeError:
-            return color_spec
+        except AttributeError as e:
+            msg = f"There is no color definition named {color_spec}"
+            raise AttributeError(msg) from e
+        if callable(ret):
+            return np.asarray(ret())
+        return np.asarray(ret)
 
     @property
     def corners(self) -> list:
+        """
 
-        '''
-        Returns lat and lon of lower left (ll) and upper right(ur) corners:
+        Returns lat and lon of lower left (ll) and upper right(ur) corners.
+
+        Order:
                ll_lat, ur_lat, ll_lon, ur_lon
-        '''
+        """
 
         lat, lon = self.latlons()
-        if self.model in ['global', 'hfip', 'obs']:
-            ret = [lat[-1], lat[0], lon[0], lon[-1]]
-        elif self.model == 'global_mpas':
-            ret = [lat[0], lat[-1], lon[0], lon[-1]]
-        else:
-            ret = [lat[0, 0], lat[-1, -1], lon[0, 0], lon[-1, -1]]
+        if len(lat.shape) == 2:
+            return [
+                np.round(x, decimals=6) for x in [lat[0, 0], lat[-1, -1], lon[0, 0], lon[-1, -1]]
+            ]
+        return [np.round(x, decimals=6) for x in [lat[0], lat[-1], lon[0], lon[-1]]]
 
-        return ret
+    @property
+    def data(self) -> DataArray:
+        """
+        Sets the data property on the object for use when we need to update
+        the values associated with a given object -- helpful for differences.
+        """
+        if not hasattr(self, "_data"):
+            self._data = self.values()
+        return self._data
```

Comment:
```suggestion
    @cached_property
    def data(self) -> DataArray:
        """
        Sets the data property on the object for use when we need to update
        the values associated with a given object -- helpful for differences.
        """
        return self.values()
```
Alternatively, if you don't explicitly need `self._data` to exist for other reasons. If you just need `.data` to be a reference to `.values`, I think this will do it. You'd also need `from functools import cached_property`.